geth/params/defaults.go

```diff
@@ -17,6 +19,18 @@ const (
 	// to be started along with a node.
 	RPCEnabledDefault = false
 
+	// DelayCycleForSyncStart sets the timeout for checking state of synchronization starting.
+	DelayCycleForSyncStart = 100 * time.Millisecond
+
+	// DelayCycleForSyncStartMaxWait sets the timeout synchronization must be allowed to take before we give up.
+	DelayCycleForSyncStartMaxWait = 30 * time.Second
+
+	// DelayCycleForSyncStart sets the timeout for checking state of synchronization completion.
+	DelayCycleForSyncCompleted = 1 * time.Second
+
+	// DelayCycleForSyncStartMaxWait sets the timeout synchronization must be completed to take before we give up.
+	DelayCycleForSyncCompletedMaxWait = 1 * time.Hour
```

Comment:
One more thing came to my mind. I am not sure we should enforce max time for syncing. I can imagine that syncing for a slow device can take even more for the first time and that's ok. On the other hand, no one will wait 1 hour when running a test. Having said that, maybe we shouldn't enforce max time here but let the consumer of `Node` interface decide how it's willing to wait.

Maybe something like this?

```go
select {
case <-node.DoneSync():
  if err := node.ErrSync(); err != nil {
    return fmt.Errorf("failed to sync: %s", err)
  }
case <-time.After(5 * time.Minute):
  return errors.New("syncing too long")
}
```

Reply:
Hmmm, Yes, I find this more suitable, will make the change.

Reply:
But we must still enforce max time to know if started, else we can not guarantee that sync event started. Your thoughts?

Reply:
This is a good idea but we currently are using a polling strategy, and to do this properly, we will need to make the change on `Node` in our ethereum fork.